src/node/module/file/File.js

```diff
@@ -44,21 +38,27 @@ class File extends FileModule {
     });
   }
   /**
-     * Upload file from local to kintone environment
-     * @param {Object} params
-     * @param {String} params.filePath
-     * @param {String} params.fileName
-     * @param {String} params.fileContent
-     * @return {Promise}
-     */
-  upload({filePath, fileName, fileContent} = {}) {  
+   * Upload file from local to kintone environment
+   * @param {Object} params
+   * @param {String} params.filePath
+   * @param {String} params.fileName
+   * @param {String} params.fileContent
+   * @return {Promise}
+   */
+  upload({filePath, fileName, fileContent} = {}) {
     let validFilecontent = fileContent;
     let validFilename = fileName;
-    if (filePath) {
-      validFilecontent = fs.createReadStream(filePath);
-      validFilename = path.basename(filePath);
-    }
-    return super.upload({fileName: validFilename, fileContent: validFilecontent});
+    return new Promise((resolve, reject) => {
+      if (filePath) {
+        try {
+          validFilecontent = fs.createReadStream(filePath);
+          validFilename = path.basename(filePath);
+        } catch (error) {
+          reject(new KintoneAPIException(error.message, error));
+        }
+      }
+      return resolve(super.upload({fileName: validFilename, fileContent: validFilecontent}));
+    });
```

Comment:
I think this snippet should be
```javascript
return new Promise((resolve, reject) => {
  if (filePath) {
    try {
      validFilecontent = fs.createReadStream(filePath);
      validFilename = path.basename(filePath);
      resolve(super.upload({fileName: validFilename, fileContent: validFilecontent}));
    } catch (error) {
      reject(new KintoneAPIException(error.message, error));
    }
  }
});
```
No need to `return` the `resolve` functions, and `resolve` should be called in side `try` block too.